src/main/java/mate/academy/spring/controller/MovieSessionController.java

```diff
@@ -0,0 +1,63 @@
+package mate.academy.spring.controller;
+
+import java.time.LocalDate;
+import java.util.List;
+import java.util.stream.Collectors;
+import mate.academy.spring.dto.MovieSessionRequestDto;
+import mate.academy.spring.dto.MovieSessionResponseDto;
+import mate.academy.spring.model.MovieSession;
+import mate.academy.spring.service.MovieSessionService;
+import mate.academy.spring.service.mapper.MovieSessionMapper;
+import org.springframework.format.annotation.DateTimeFormat;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/movie-sessions")
+public class MovieSessionController {
+    private final MovieSessionService sessionService;
+    private final MovieSessionMapper sessionMapper;
+
+    public MovieSessionController(MovieSessionService sessionService,
+                                  MovieSessionMapper sessionMapper) {
+        this.sessionService = sessionService;
+        this.sessionMapper = sessionMapper;
+    }
+
+    @PostMapping
+    public MovieSessionResponseDto add(@RequestBody MovieSessionRequestDto requestDto) {
+        return sessionMapper.toDto(sessionService
+                .add(sessionMapper.toModel(requestDto)));
+    }
+
+    @GetMapping("/available")
+    public List<MovieSessionResponseDto> findAvailableSessions(
+            @RequestParam Long movieId,
+            @RequestParam @DateTimeFormat(pattern = "dd.MM.yyyy") LocalDate date) {
+        return sessionService.findAvailableSessions(movieId, date).stream()
+                .map(sessionMapper::toDto)
+                .collect(Collectors.toList());
+    }
+
+    @PutMapping("/{id}")
+    public MovieSessionResponseDto update(
+            @PathVariable Long id,
+            @RequestBody MovieSessionRequestDto requestDto) {
+        MovieSession movieSession = sessionMapper.toModel(requestDto);
+        movieSession.setId(id);
+        sessionService.update(movieSession);
+        return sessionMapper.toDto(movieSession);
+    }
+
+    @DeleteMapping("/{movieId}")
```

Comment:
```suggestion
    @DeleteMapping("/{id}")
```
You can use the same endpoint here. Since there are different operations and there will be no conflicts